src/oss-find-source/FindSourceTool.cs

```diff
@@ -139,52 +125,55 @@ public FindSourceTool() : base()
+        /// <param name="outputBuilder"> </param>
+        /// <param name="purl"> </param>
+        /// <param name="results"> </param>
+        private void AppendOutput(IOutputBuilder outputBuilder, PackageURL purl, List<KeyValuePair<PackageURL, double>> results)
         {
-            if (outputBuilder.isTextFormat())
-            {
-                outputBuilder.AppendOutput(GetTextResults(results));
-            }
-            else
+            switch (currentOutputFormat)
             {
-                outputBuilder.AppendOutput(GetSarifResults(purl, results));
+                case OutputFormat.text:
+                default:
+                    outputBuilder.AppendOutput(GetTextResults(results));
+                    break;
+
+                case OutputFormat.sarifv1:
+                case OutputFormat.sarifv2:
+                    outputBuilder.AppendOutput(GetSarifResults(purl, results));
+                    break;
             }
         }
 
         /// <summary>
-        /// Convert findSourceTool results into text format
+        ///     Convert findSourceTool results into text format
         /// </summary>
-        /// <param name="results"></param>
-        /// <returns></returns>
-        static string GetTextResults(List<KeyValuePair<PackageURL, double>> results)
+        /// <param name="results"> </param>
+        /// <returns> </returns>
+        private static List<string> GetTextResults(List<KeyValuePair<PackageURL, double>>? results)
         {
-            StringBuilder stringOutput = new StringBuilder();
-            foreach (var result in results)
+            //StringBuilder stringOutput = new StringBuilder();
+            List<string> stringOutput = new List<string>();
+            foreach (var result in results ?? new List<KeyValuePair<PackageURL, double>>())
             {
                 var confidence = result.Value * 100.0;
-                stringOutput.Append(
+                stringOutput.Add(
                     $"{confidence:0.0}%\thttps://github.com/{result.Key.Namespace}/{result.Key.Name} ({result.Key})");
-                stringOutput.Append(
-                    Environment.NewLine);
             }
-            return stringOutput.ToString();
+            return stringOutput;
         }
 
         /// <summary>
-        /// Build and return a list of Sarif Result list from the find source results
+        ///     Build and return a list of Sarif Result list from the find source results
         /// </summary>
-        /// <param name="purl"></param>
-        /// <param name="results"></param>
-        /// <returns></returns>
-        static List<Result> GetSarifResults(PackageURL purl, List<KeyValuePair<PackageURL, double>> results)
+        /// <param name="purl"> </param>
+        /// <param name="results"> </param>
+        /// <returns> </returns>
+        private static List<Result> GetSarifResults(PackageURL purl, List<KeyValuePair<PackageURL, double>>? results)
```

Comment:
This is a private API, why allow it to take null for results (no-op)?

Reply:
changed